lib/src/model.dart

```diff
@@ -3288,23 +3285,48 @@ abstract class ModelElement extends Canonicalization
     return docFrom;
   }
 
-  String _buildDocumentationLocal() => _buildDocumentationBase();
+  String _buildDocumentationLocal() => _buildDocumentationBaseSync();
+
+  /// Override this to add more features to the documentation builder in a
+  /// subclass.
+  String _buildDocumentationAddition(String docs) => docs;
 
   /// Separate from _buildDocumentationLocal for overriding.
-  String _buildDocumentationBase() {
+  String _buildDocumentationBaseSync() {
+    assert(_rawDocs == null);
+    // Do not use the sync method if we need to evaluate tools or templates.
+    assert(packageGraph._localDocumentationBuilt);
+    if (config.dropTextFrom.contains(element.library.name)) {
+      _rawDocs = '';
+    } else {
+      _rawDocs = documentationComment ?? '';
+      _rawDocs = stripComments(_rawDocs) ?? '';
+      _rawDocs = _injectExamples(_rawDocs);
+      _rawDocs = _injectAnimations(_rawDocs);
+      _rawDocs = _stripHtmlAndAddToIndex(_rawDocs);
+    }
+    _rawDocs = _buildDocumentationAddition(_rawDocs);
+    return _rawDocs;
+  }
+
+  /// Separate from _buildDocumentationLocal for overriding.  Can only be
+  /// used as part of [PackageGraph.setUpPackageGraph].
+  Future<String> _buildDocumentationBase() async {
     assert(_rawDocs == null);
+    // Do not use the sync method if we need to evaluate tools or templates.
     if (config.dropTextFrom.contains(element.library.name)) {
       _rawDocs = '';
     } else {
       _rawDocs = documentationComment ?? '';
       _rawDocs = stripComments(_rawDocs) ?? '';
       // Must evaluate tools first, in case they insert any other directives.
-      _rawDocs = _evaluateTools(_rawDocs);
+      _rawDocs = await _evaluateTools(_rawDocs);
```

Comment:
Does dartdoc have the unawaited futures lint enabled? It might be useful in this codebase, given that several previously sync methods became async ones.